javascript/ql/src/semmle/javascript/dataflow/TaintTracking.qll

```diff
@@ -267,6 +267,12 @@ module TaintTracking {
       pred = DataFlow::valueNode(fos.getIterationDomain()) and
       succ = DataFlow::lvalueNode(fos.getLValue())
     )
+    or
+    // rest-pattern inside a propety pattern. E.g. from `foo:..` to `args` in `const {foo: {...args}} = something()`.
+    exists(PropertyPattern pattern |
+      pred.(DataFlow::PropNode).getAstNode() = pattern and
+      succ = DataFlow::lvalueNode(pattern.getValuePattern().(ObjectPattern).getRest())
+    )
```

Comment:
Have you tried the version I proposed earlier?
```suggestion
    exists(ObjectPattern pattern |
      pred = lvalueNode(pattern) and
      succ = lvalueNode(pattern.getRest())
    )
```

That should also work for rest patterns that aren't part of a property pattern, like:
```js
let { ...args } = x
let [ { ...args} ] = x;
```

Reply:
I had not, I should have.  

I've implemented your suggest, but I used `DestructuringPattern` instead of `ObjectPattern`, that way we also support array-spreads.